src/Microsoft.AspNet.Mvc.Core/Filters/ResponseCacheAttribute.cs

```diff
@@ -38,7 +39,17 @@ public int Duration
         /// <summary>
         /// Gets or sets the location where the data from a particular URL must be cached.
         /// </summary>
-        public ResponseCacheLocation Location { get; set; }
+        public ResponseCacheLocation Location
```

Comment:
is this change related? If so just make Location { get; set; } = ResponseCacheLocation.Any. and throw on setting null.